packages/editor/src/components/EditorContainer.tsx

```diff
@@ -177,12 +178,12 @@ const onSaveAs = async () => {
       })
       DialogState.setDialog(null)
       if (result?.name && projectName) {
-        await saveScene(projectName, result.name, abortController.signal)
+        await saveSceneJSON(sceneAssetID, projectName, result.name, abortController.signal)
         getMutableState(SceneState).sceneModified.set(false)
-        const newSceneData = (await Engine.instance.api
+        const [newSceneData] = (await Engine.instance.api
           .service(scenePath)
-          .get('', { query: { project: projectName, name: result.name, metadataOnly: true } })) as SceneDataType
-        setSceneInState(newSceneData.scenePath)
+          .find({ query: { assetURL: getState(EditorState).scenePath! } })) as any as AssetDataType[]
```

Comment:
Wouldnt it be a Paginated object? Because if you want an array, you need to pass `paginate: false` in query

Reply:
good catch!